src/org/zaproxy/zap/extension/spider/SpiderAPI.java

```diff
@@ -460,6 +466,29 @@ public ApiResponse handleApiView(String name, JSONObject params) throws ApiExcep
 				resultList.addItem(new ApiResponseSet("scan", map));
 			}
 			result = resultList;
+		} else if (VIEW_ALL_URLS.equals(name)) {
+	        ExtensionHistory extHistory = ((ExtensionHistory)Control.getSingleton().getExtensionLoader().getExtension(ExtensionHistory.NAME));
+
+			if (extHistory == null) {
+				throw new ApiException(Type.NO_IMPLEMENTOR, ExtensionHistory.NAME);
+			}
+	        
+			ApiResponseList resultUrls = new ApiResponseList(name);
+			Set<String> urlSet=new HashSet<String>();
+			
+			for (int idx = 0; idx <= extHistory.getLastHistoryId(); idx++) {
+				HistoryReference hr=extHistory.getHistoryReference(idx);
```

Comment:
This requires to read (and caching almost) all HTTP messages, even if they are not of interest. It could obtain the spider HTTP messages directly, for example:
```Java
TableHistory tableHistory = extension.getModel().getDb().getTableHistory();
List<Integer> ids = tableHistory.getHistoryIdsOfHistType(extension.getModel().getSession().getSessionId(), HistoryReference.TYPE_SPIDER, HistoryReference.TYPE_SPIDER_TASK);
for (Integer id : ids) {
    String uri = tableHistory.read(id.intValue()).getHttpMessage().getRequestHeader().getURI().toString();
    // ...
```